include/deal.II/base/utilities.h

```diff
@@ -730,9 +770,107 @@ namespace Utilities
           len = half;
       }
   }
+
+
+
+  template<typename T>
+  std::vector<char> pack(const T &object)
+  {
+    // set up a buffer and then use it as the target of a compressing
+    // stream into which we serialize the current object
+    std::vector<char> buffer;
+    {
+#ifdef DEAL_II_WITH_ZLIB
+      boost::iostreams::filtering_ostream out;
+      out.push(boost::iostreams::gzip_compressor
+               (boost::iostreams::gzip_params
+                (boost::iostreams::gzip::best_compression)));
+      out.push(boost::iostreams::back_inserter(buffer));
+
+      boost::archive::binary_oarchive archive(out);
+      archive << object;
+      out.flush();
+#else
+      std::ostringstream out;
+      boost::archive::binary_oarchive archive(out);
+      archive << object;
+      const std::string &s = out.str();
+      buffer.reserve(s.size());
+      buffer.assign(s.begin(), s.end());
+#endif
+    }
+    return buffer;
+  }
+
+
+
+  template<typename T>
+  T unpack(const std::vector<char> &buffer)
+  {
+    std::string decompressed_buffer;
+    T object;
+
+    // first decompress the buffer
+    {
+#ifdef DEAL_II_WITH_ZLIB
+      boost::iostreams::filtering_ostream decompressing_stream;
+      decompressing_stream.push(boost::iostreams::gzip_decompressor());
+      decompressing_stream.push(boost::iostreams::back_inserter(decompressed_buffer));
+      decompressing_stream.write (buffer.data(), buffer.size());
+#else
+      decompressed_buffer.assign (buffer.begin(), buffer.end());
+#endif
+    }
+
+    // then restore the object from the buffer
+    std::istringstream in(decompressed_buffer);
+    boost::archive::binary_iarchive archive(in);
+
+    archive >> object;
+    return object;
+  }
 }
 
 
 DEAL_II_NAMESPACE_CLOSE
 
+#ifndef DOXYGEN
+namespace boost
+{
+  namespace serialization
+  {
+
+    // Provides boost and c++11 with a way to serialize tuples and pairs automatically
```

Comment:
do you mind adding tests for those?

Reply:
maybe put it into a separate PR so that this one can be merged quickly

Reply:
nevermind, i see that you already pushed the tests.

Reply:
it's already there. test number two packs and unpack a tuple of pairs and points.